rest/data/project.go

```diff
@@ -389,3 +389,54 @@ func (pc *DBProjectConnector) GetProjectFromFile(ctx context.Context, pRef model
 	}
 	return model.GetProjectFromFile(ctx, opts)
 }
+
+func DeleteBranch(ctx context.Context, pRef *model.ProjectRef) error {
+	if pRef.IsHidden() {
+		return gimlet.ErrorResponse{
+			StatusCode: http.StatusBadRequest,
+			Message:    errors.Errorf("project '%s' is already hidden", pRef.Id).Error(),
+		}
+	}
+
+	if !pRef.UseRepoSettings() {
+		return gimlet.ErrorResponse{
+			StatusCode: http.StatusBadRequest,
+			Message:    errors.Errorf("project '%s' must be attached to a repo to be eligible for deletion", pRef.Id).Error(),
```

Comment:
I have no idea why this is true. Can we make a follow up ticket for app to investigate this?

Reply:
Opened [EVG-19958](https://jira.mongodb.org/browse/EVG-19958)!